specification/confidentialledger/Microsoft.mCCF/proposals.tsp

```diff
@@ -0,0 +1,164 @@
+  @doc("Proposal is active and can be voted on.")
+  Open,
+
+  @doc("Proposal passed a successful vote and was enacted.")
+  Accepted,
+
+  @doc("Proposal was removed by proposing member. Will never be enacted.")
+  Withdrawn,
+
+  @doc("Proposal was rejected by vote. Will never be enacted.")
+  Rejected,
+
+  @doc("Proposal passed a successful vote, but its proposed actions failed. Will never be enacted.")
+  Failed,
+
+  @doc("Proposal was open when its semantics were potentially changed (code or constitution were modified), so it was automatically invalidated.")
+  Dropped,
+}
+
+@doc("Description of why governance execution failed.")
+model FailureInfo {
+  @doc("Error message describing reason for failure.")
+  reason: string;
+
+  @doc("Call stack showing where failure occurred, if available.")
+  trace?: string;
+}
+
+@doc("Each key is a memberId, and the corresponding value is the result of their ballot.")
+model MemberVotes is Record<boolean>;
+
+@doc("Each key is a memberId, and the corresponding value explains why execution of their ballot failed.")
+model VoteFailures is Record<FailureInfo>;
+
+// Corresponds to ccf::jsgov::ProposalInfoSummary in C++
+@doc("Description of a proposed change to the service.")
+@TypeSpec.Rest.resource("proposals")
+model Proposal {
+  @key
+  @doc("Unique ID assigned to this proposal at its submission, by the service.")
+  @visibility("read")
+  proposalId: proposalId;
+
+  @doc("ID of CCF member who signed and created this proposal.")
+  @visibility("read")
+  proposerId: memberId;
+
+  @doc("Current state of this proposal.")
+  proposalState: ProposalState;
+
+  @doc("Count of how many ballots have been submitted for this proposal.")
+  ballotCount: int64;
+
+  @doc("If a proposal is not open, then this contains the result of each ballot when the proposal transitioned from open.")
+  finalVotes?: MemberVotes;
+
+  @doc("This will be populated with a description of any ballots which failed to execute.")
+  voteFailures?: VoteFailures;
+
+  @doc("If proposalState is Failure, then this will describe why the proposal failed.")
+  failure?: FailureInfo;
+}
+
+@doc("The actions contained in a proposal, describing the proposal's changes.")
+@TypeSpec.Rest.parentResource(Proposal)
+model ProposalActions {
+  @doc("A list of actions to apply. Each action is considered, in order, for both proposal and execution of the proposal. All actions are validated and, if the proposal is accepted, applied atomically.")
+  actions: Action[];
+}
+
+@doc("The source script of an executable vote from a member, regarding a proposed change to the service.")
+@TypeSpec.Rest.resource("ballots")
+@TypeSpec.Rest.parentResource(Proposal)
+model Ballot {
+  @doc("ID of CCF member who signed and submitted this ballot.")
+  @visibility("read")
+  @key
+  memberId: memberId;
+
+  @doc("The script to execute, returning the voter's support of this proposal.")
+  @visibility("create")
+  @projectedName("csharp", "BallotScript")
```

Comment:
Similar question as before, but I'll not comment on this again for brevity. Please evaluation all locations where you're changing just C#. Either project the name for all languages, or consider changing the model name.